File: server/database/src/projections/spot/asset.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::types::Uuid;

use crate::types::Fraction;

#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
pub struct Asset {
    pub id: Uuid,
    pub created_at: DateTime<Utc>,
    pub last_modification_at: DateTime<Utc>,
    pub name: String,
    pub symbol: String,
    pub maker_fee: Fraction,
    pub taker_fee: Fraction,
}

File: web/app/src/types/asset.ts
import { z } from "zod";
import { Fraction } from "~/types/primitives/fraction";
import { Datetime } from "./primitives/datetime";
import { Uuid } from "./primitives/uuid";

export const Asset = z.object({
  id: Uuid,
  created_at: Datetime,
  name: z.string(),
  symbol: z.string(),
  maker_fee: Fraction,
  taker_fee: Fraction,
});
export type Asset = z.infer<typeof Asset>;
